twitter/searcher.py

```diff
@@ -0,0 +1,182 @@
+import tweepy
+from typing import Dict, List, Any, Set, Tuple
+from datetime import datetime
+import json
+
+# from boto3 import client as boto_client
+
+from common.data_classes import RawSubmission
+from common.enums import DataSource
+
+from clients.laravel_client import bulk_upload_submissions
+from common.config import (
+    TWITTER_LAST_RUN_FILENAME,
+    # TWITTER_LAST_RUN_BUCKET,
+    TWITTER_LARAVEL_API_KEY,
+    TWITTER_CONSUMER_KEY,
+    TWITTER_CONSUMER_SECRET,
+    TWITTER_ACCESS_TOKEN,
+    TWITTER_ACCESS_TOKEN_SECRET,
+)
+
+# from common.utils import write_raw_submissions_to_csv
+
+QUERIES = [
+    # recommended by freezman
+    "#bluefall",
+    "#PoliceBrutality",
+    "#PoliceBrutailtyPandemic",
+    "#Protests2020",
+    # cities by population - protests
+    "#nycProtests",
+    "#newyorkProtests",
+    "#losangelesProtests",
+    "#laprotests",
+    "#stLouisProtests",
+    "#stlProtests",
+    "#philadelphiaprotests",
+    "#phillyprotests",
+    "#chicagoprotests",
+    "#houstonprotests",
+    "#pheonixProtests",
+    "#miamiProtests",
+    "#dcprotests",
+    "#WashingtonDCProtest",
+    "#seattleprotest",
+    "#austinprotest"
+    # found from 949mac's endpoint https://api.846policebrutality.com/api/incidents?include=evidence
+    "#GeorgeFloyd",
+    "#JusticeForGeorgeFloyd",
+    "#AbolishThePolice",
+    "#BlackLivesMatter",
+    "tear gas",
+    # Common occurrences
+    "#DefundThePolice",
+]
+
+
+def run_twitter_searches(since_id: int) -> int:
+    if not since_id:
+        since_id = get_since_id_from_file()
+    api = build_tweepy_api()
+    # tweets = []
+    total_returned_tweets = 0
+    processed_id_tweets = set()
+    max_processed_id = 0
+    max_processed_time_stamp = 0
+    for query in QUERIES:
+        cursor = query_twitter(api, query, since_id)
+        for resp in cursor:
+            total_returned_tweets += 1
+            tweet = resp._json
+            id_tweet = tweet["id"]
+            if id_tweet > max_processed_id:
+                max_processed_id = id_tweet
+                max_processed_time_stamp = tweet["created_at"]
+                # TODO: should we keep updating this file, so it is up to date if the job quits unexpectedly?
```

Comment:
I think log_last_processed_id is plenty. Unless you wanna get serious and have a last processed id for every query. But that seems like overkill

Reply:
yea my thoughts exactly. prob better to improve dupe detection for laravel if it comes down to it